### PR TITLE
Handle 429 from auto-model service

### PR DESCRIPTION
**Pull request by charisk**
Shows a slightly more helpful message when we receive 429s from the auto-model endpoint.

## Checklist
N/A - internal only:
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts [hunk 1]
```diff
@@ -479,4 +483,19 @@ export class DataExtensionsEditorView extends AbstractWebview<
       message: "",
     });
   }
+
+  private async callApi(request: ModelRequest): Promise<ModelResponse | null> {
```
Comment (koesie10) [inline extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts:487]: Can we make this name a little less generic? The data extensions editor isn't only for auto-modeling, so it's not clear from the name which API this is calling.

File: extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts [hunk 2]
```diff
@@ -479,4 +483,19 @@ export class DataExtensionsEditorView extends AbstractWebview<
...
+    try {
+      return await autoModel(this.app.credentials, request);
+    } catch (e) {
+      if (e instanceof RequestError && e.status === 429) {
+        void showAndLogExceptionWithTelemetry(
+          redactableError`Rate limit hit, please try again soon.`,
```
Comment (koesie10) [inline extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts:493]: We should probably also report the full error:

```suggestion
          redactableError(e)`Rate limit hit, please try again soon.`,
```

File: extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts (hunk 1 above)
Comment (charisk) [reply extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts]: 🤦‍♀️

**Review by koesie10 — APPROVED** [review]
LGTM, although the progress doesn't get cleared when I run into a rate limit error. I think you'll need to add `await this.clearProgress();` just before `return;` in the `generateModeledMethodsFromLlm` method.

**Comment by charisk** [comment]
> LGTM, although the progress doesn't get cleared when I run into a rate limit error. I think you'll need to add `await this.clearProgress();` just before `return;` in the `generateModeledMethodsFromLlm` method.

Ah, I was contemplating whether to tidy this up in this PR. I've added `this.clearProgress()` in the catch block to make sure we clear on other failures. too.

**Comment by koesie10** [comment]
> Ah, I was contemplating whether to tidy this up in this PR. I've added `this.clearProgress()` in the catch block to make sure we clear on other failures. too.

Thanks! We might want to move that into a `catch` block inside the `generateModeledMethodsFromLlm` method to also catch errors when the query fails, but this is error is much more likely to happen, so that can happen in a follow-up.

**Pull request merged**